src/extensions/agent_modules/shared/mosquitto_utils/src/adu_mosquitto_utils.c

```diff
@@ -140,7 +146,25 @@ bool ADU_are_correlation_ids_matching(
 
     if (outCorrelationData != NULL)
     {
-        if (ADUC_AllocAndStrCopyN(outCorrelationData, responseCorrelationId, (size_t)responseCorrelationIdLen) != 0)
+        // TODO: Remove this workaround once adu service deploys fix that ensures exact correlation id string agent sent is
```

Comment:
reviewing now -- line 140 seems really dangerous and potential for a 0-day.  if read_binary returns binary data, how do we know:

1. that it's a string
2. that it's null-terminated?

Reply:
Yes, good catch!  

To properly validate, it needs to be verifying that the bytes patterns are those of valid utf-8 encodings. Also, after such verification, it cannot assume the utf-8 is null-terminated. However, we /are/ given the byte length (`responseCorrelationIdLen`) so the latter could be avoided.

I'm going to side-step the copying and output parameters because it's only needed for tracing to logs. Instead, I will replace with a trace (and add comments not to trace the binary data that could possibly NOT be utf-8) with:

a static trace that just says the trusted `expectededCorrelationId` string did not match the untrusted binary data correlation-data.